HLTrigger/HLTfilters/plugins/HLTTripletMass.h

```diff
@@ -0,0 +1,244 @@
+
+public:
+  explicit HLTTripletMass(const edm::ParameterSet&);
+  ~HLTTripletMass() override = default;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  bool hltFilter(edm::Event&,
+                 const edm::EventSetup&,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override;
+  bool getCollections(edm::Event& iEvent,
+                      std::vector<T1Ref>& coll1,
+                      std::vector<T2Ref>& coll2,
+                      std::vector<T3Ref>& coll3,
+                      trigger::TriggerFilterObjectWithRefs& filterproduct) const;
+
+private:
+  // configuration
+  const std::vector<edm::InputTag> originTag1_;  // input tag identifying originals 1st product
+  const std::vector<edm::InputTag> originTag2_;  // input tag identifying originals 2nd product
+  const std::vector<edm::InputTag> originTag3_;  // input tag identifying originals 3rd product
+  const edm::EDGetTokenT<trigger::TriggerFilterObjectWithRefs> inputToken1_;
+  const edm::EDGetTokenT<trigger::TriggerFilterObjectWithRefs> inputToken2_;
+  const edm::EDGetTokenT<trigger::TriggerFilterObjectWithRefs> inputToken3_;
+  const int triggerType1_;
+  const int triggerType2_;
+  const int triggerType3_;
+  const std::vector<double> min_InvMass_;  // minimum invariant mass of pair
+  const std::vector<double> max_InvMass_;  // maximum invariant mass of pair
+  const double max_DR_;                    // maximum deltaR between the p4 of 3rd product and p4 of (1+2) product
+  const double max_DR2_;                   // maximum deltaR^2 between the p4 of 3rd product and p4 of (1+2) product
+  const int min_N_;
+  const bool is1and2Same;
+  const bool is2and3Same;
+};
+
+template <typename T1, typename T2, typename T3>
+HLTTripletMass<T1, T2, T3>::HLTTripletMass(const edm::ParameterSet& iConfig)
+    : HLTFilter(iConfig),
+      originTag1_(iConfig.getParameter<std::vector<edm::InputTag>>("originTag1")),
+      originTag2_(iConfig.getParameter<std::vector<edm::InputTag>>("originTag2")),
+      originTag3_(iConfig.getParameter<std::vector<edm::InputTag>>("originTag3")),
+      inputToken1_(consumes(iConfig.getParameter<edm::InputTag>("inputTag1"))),
+      inputToken2_(consumes(iConfig.getParameter<edm::InputTag>("inputTag2"))),
+      inputToken3_(consumes(iConfig.getParameter<edm::InputTag>("inputTag3"))),
+      triggerType1_(iConfig.getParameter<int>("triggerType1")),
+      triggerType2_(iConfig.getParameter<int>("triggerType2")),
+      triggerType3_(iConfig.getParameter<int>("triggerType3")),
+      min_InvMass_(iConfig.getParameter<vector<double>>("MinInvMass")),
+      max_InvMass_(iConfig.getParameter<vector<double>>("MaxInvMass")),
+      max_DR_(iConfig.getParameter<double>("MaxDR")),
+      max_DR2_(max_DR_ * max_DR_),
+      min_N_(iConfig.getParameter<int>("MinN")),
+      is1and2Same(iConfig.getParameter<bool>("is1and2Same")),
+      is2and3Same(iConfig.getParameter<bool>("is2and3Same")) {
```

Comment:
```suggestion
      is1and2Same_(iConfig.getParameter<bool>("is1and2Same")),
      is2and3Same_(iConfig.getParameter<bool>("is2and3Same")) {
```
I missed this.. these should have an underscore. I would be in favour of fixing this small thing in a future PR, since this PR is targeting the next pre-release.